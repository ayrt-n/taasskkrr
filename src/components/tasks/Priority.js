import React from 'react';
import '../../styles/Tasks.css'

function Priority({ priorityEnum }) {
  let priority = ''

  if (priorityEnum === 0) {
    priority = 'Low'
  } else if (priorityEnum === 1) {
    priority = 'Medium'
  } else {
    priority = 'High'
  }

  return (
    <div className={`Task-priority ${priority}`}>
      <svg style={{width:"1rem", height: "1rem"}} viewBox="0 0 24 24">
        <path fill="currentColor" d="M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z" />
      </svg>
      <p>{priority}</p>
    </div>
  );
}

export default Priority;
