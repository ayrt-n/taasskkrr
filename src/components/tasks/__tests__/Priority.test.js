import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Priority from '../Priority';

describe('Priority component', () => {
  describe('when given different priority enums', () => {
    it('correctly renders low priority', () => {
      const { container } = render(<Priority priorityEnum={0} />)
      expect(container).toMatchSnapshot();
    });

    it('correctly renders medium priority', () => {
      const { container } = render(<Priority priorityEnum={1} />)
      expect(container).toMatchSnapshot();
    });

    it('correctly renders high priority', () => {
      const { container } = render(<Priority priorityEnum={2} />)
      expect(container).toMatchSnapshot();
    });
  })
});
