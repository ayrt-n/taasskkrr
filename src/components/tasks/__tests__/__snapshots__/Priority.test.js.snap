// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Priority component when given different priority enums correctly renders high priority 1`] = `
<div>
  <div
    class="Task-priority High"
  >
    <svg
      style="width: 1rem; height: 1rem;"
      viewBox="0 0 24 24"
    >
      <path
        d="M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
        fill="currentColor"
      />
    </svg>
    <p>
      High
    </p>
  </div>
</div>
`;

exports[`Priority component when given different priority enums correctly renders low priority 1`] = `
<div>
  <div
    class="Task-priority Low"
  >
    <svg
      style="width: 1rem; height: 1rem;"
      viewBox="0 0 24 24"
    >
      <path
        d="M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
        fill="currentColor"
      />
    </svg>
    <p>
      Low
    </p>
  </div>
</div>
`;

exports[`Priority component when given different priority enums correctly renders medium priority 1`] = `
<div>
  <div
    class="Task-priority Medium"
  >
    <svg
      style="width: 1rem; height: 1rem;"
      viewBox="0 0 24 24"
    >
      <path
        d="M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
        fill="currentColor"
      />
    </svg>
    <p>
      Medium
    </p>
  </div>
</div>
`;
